cdc/server.go

```diff
@@ -258,17 +258,11 @@ func (s *Server) etcdHealthChecker(ctx context.Context) error {
 			for _, pdEndpoint := range s.pdEndpoints {
 				start := time.Now()
 				ctx, cancel := context.WithTimeout(ctx, time.Second*10)
-				req, err := http.NewRequestWithContext(
-					ctx, http.MethodGet, fmt.Sprintf("%s/pd/api/v1/health", pdEndpoint), nil)
-				if err != nil {
-					log.Warn("etcd health check failed", zap.Error(err))
-					cancel()
-					continue
-				}
-				_, err = httpCli.Do(req)
+				resp, err := httpCli.Get(ctx, fmt.Sprintf("%s/pd/api/v1/health", pdEndpoint))
 				if err != nil {
 					log.Warn("etcd health check error", zap.Error(err))
 				} else {
+					_ = resp.Body.Close()
```

Comment:
It's better to discard the response body to reuse the underlying TCP connection.

Reply:
Is there any reference?

Reply:
https://github.com/golang/go/blob/4d8db00641cc9ff4f44de7df9b8c4f4a4f9416ee/src/net/http/response.go#L64